docs/guidelines.md

```diff
@@ -0,0 +1,130 @@
+are to be interpreted as in [RFC2119](https://www.ietf.org/rfc/rfc2119.txt);
+given that we recognise the following four output states and their restrictions:
+
+- `approve_as_optimal`: **SHOULD** be a best solution, **SHOULD** be without
+  comment
+- `approve_with_comment`: **COULD** be a best solution, **MUST** be with comment
+- `disapprove_with_comment`: **MUST** be with comment
+- `refer_to_mentor`: **COULD** be with comment
+
+### Why is it not ... (e.g. **MUST**)?
+
+Per [RFC2119](https://www.ietf.org/rfc/rfc2119.txt), if **MUST** is used, it is
+a guarantee that the rule is always so, and does not need to be guarded for. For
+example, **MUST** be without comment means that the website could crash if an
+analyzer sends a comment anyway. **SHOULD** indicates any consumer of the output
+must still guard against unwanted behaviour.
+
+### Idiomatic rules / Language features / Stylistic choices
+
+In this document, **de facto** is defined as follows:
+
+- **de facto**: describes practices that exist in reality, even if not
+  officially recognized by laws.
+- **de facto standard**: a custom or convention that has achieved a dominant
+  position by public acceptance or market forces. Unofficial customs that are
+  widely accepted.
+
+In other words, if a nearly all developers (non-hobyists) who write code in a 
+certain language have established certain rules, these rules are a **de facto 
+standard** and become idiomatic use. Example: **Ruby** uses 2 space identation.
+
+Some rules are language features, even if they are not documented well. These
+language features are part of "idiomatic rules" and not stylistic choices.
+Example: **Ruby**'s MRI treats variables named `_` differently.
+
+Finally there are rules that are pure preferences, even though they might be
+adopted by large bodies such as organisations and corporations. These rules
+are usually part of _competing_ standards. Exercism does not favour one over
+another. Example: **TypeScript** has a linter `tslint` (or `eslint` + plugin)
+which is maintained by a company that is not Microsoft. It competes with other
+linters such as `xo`. Most of the rules are not language features or idiomatic
+rules, and therefore stylistic choices.
+
+## Conditions and outcome
+
+For feedback generated by automated mentoring:
+
+- it **SHOULD** `disapprove_with_comment` when someone isn't using one of the
+  subjects the exercise is supposed to teach,
+- each `disapprove_with_comment` **MUST** have at least one comment,
+- each `disapprove_with_comment` **SHOULD** steer a student towards an
+  `approve_as_optimal`,
+- a student **SHOULD NOT** get a `refer_to_mentor` in a newer iteration, if
+  they've had a `disapprove_with_comment` in a previous iteration,
+- a `refer_to_mentor` **COULD** have one or more comments which will be given to
+  the mentor.
+
+There is no minimum or maximum amount of :speech_balloon: comments. However,
```

Comment:
Maybe :)

I'd maybe say "It is currently undecided as to whether there is a maximum number of comments. Our recommendation is currently to aim for one comment per iteration" (because that's my gut instinct, although far from certain).

Reply:
Agreed!